src/jsp/com/cubrid/jsp/LoggingThread.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *
+ * Copyright (c) 2016 CUBRID Corporation.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * - Neither the name of the <ORGANIZATION> nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ *
+ */
+
+package com.cubrid.jsp;
+
+import java.io.IOException;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.logging.FileHandler;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class LoggingThread extends Thread {
+    private final Logger logger = Logger.getLogger("com.cubrid.jsp");
+
+    private FileHandler logHandler = null;
+    private LinkedBlockingQueue<String> logQueue = new LinkedBlockingQueue<String>();
+    private Level logginLevel = Level.SEVERE;
+
+    public LoggingThread(String path) throws SecurityException, IOException {
+        super();
+        logHandler = new FileHandler(path, true);
+        logger.addHandler(logHandler);
+    }
+
+    @Override
+    public void run() {
+        while (Thread.interrupted() == false) {
+            try {
+                String logString = logQueue.take();
+                logger.log(logginLevel, logString);
+            } catch (InterruptedException e) {
+                break;
+            }
+        }
+
+        if (logHandler != null) {
+            try {
+                logHandler.close();
```

Comment:
If the queue is not empty, shouldn't we print it out and close()?

Reply:
This line can be reached in the exit routine before the server exits via exit(0). At this time, all other threads have already terminated, so the producer of the log no longer exists.